### PR TITLE
Fixes callback in Scroll To script

### PR DESCRIPTION
**Pull request by tjdunklee**
## What does this do/fix?

While working on another project, I noticed the callback for the `scroll-to.js` script had a typo in it. This fixes that callback, but also uses it to add auto element focusing to the `smooth-anchors.js` script. For accessibility reasons, we should change the focus for users if we are scrolling them to a certain position.

## Tests

Does this have tests?

- [ ] Yes
- [x] No, this doesn't need tests because this is optional functionality.
- [ ] No, I need help figuring out how to write the tests.

### CODE REVIEW

File: wp-content/themes/core/assets/js/src/utils/dom/smooth-anchors.js
Comment (dpellenwood) [inline wp-content/themes/core/assets/js/src/utils/dom/smooth-anchors.js:23]: This is great, except if the element being scrolled to isn't focusable. (such as adding an id attribute to an element which isn't a link or input, etc)  I [think this is a reasonable solution](https://github.com/moderntribe/codet-vision/blob/main/wp-content/themes/core/assets/js/src/utils/dom/smooth-anchors.js#L19-L27) to the issue. Happy to get other's input as well.

**Review by dpellenwood — APPROVED** [review]
Just one comment about focusing on the target. Thanks, Mike!

File: wp-content/themes/core/assets/js/src/utils/dom/scroll-to.js
Comment (ryanurban) [inline wp-content/themes/core/assets/js/src/utils/dom/scroll-to.js:36]: Good catch ❤️

File: wp-content/themes/core/assets/js/src/utils/dom/smooth-anchors.js
Comment (ryanurban) [reply wp-content/themes/core/assets/js/src/utils/dom/smooth-anchors.js:23]: Agreed, we should have this probably just be an additional bool argument you can opt into

**Review by ryanurban — APPROVED** [review]

**Comment by tjdunklee** [comment]
Thank @ryanurban and @dpellenwood for the great suggestions. I have an idea that will solve both of them, I'll make an update soon and have you re-review.

**Comment by tjdunklee** [comment]
I've added a boolean option called "auto_focus" that is off by default when you init the Smooth Anchors script. If it is set to TRUE, I took inspiration from Dave's Codet code to apply a tab-index to make any element focusable before setting the focus. Let me know what you think of this solution.

File: wp-content/themes/core/assets/js/src/utils/dom/smooth-anchors.js
Comment (ryanurban) [inline wp-content/themes/core/assets/js/src/utils/dom/smooth-anchors.js:28]: I just so happen to be digging into browser skip link issues so this matter is top of mind at the moment 😂

Fwiw, I was just reminded this utility only works on anchors that serve as an anchor hash link, so we don't need the tabindex functionality, and I'm going to commit the a11y tweaks I'm making to handle broken skip links, which I think will handle the focus bit.

Let's tag team this, I'll commit my work in this branch (still working on it, will have it wrapped today), you comment out and/or remove everything but the fixed callback, and then let's both test and see where we land on this? Thanks for noticing this and bringing it up!

Comment (tjdunklee) [reply wp-content/themes/core/assets/js/src/utils/dom/smooth-anchors.js:28]: Sounds good, we can definitely tag team this.

I'm a little confused though why you wouldn't need the tabindexing. The smooth scroll's destination could be anything with an `id` so like Dave mentioned it could be a non-focusable element. I'm picturing an author linking in a nav menu item to a specific block on that page.

Either way, I'll wait for your commit and make tweaks to mine based on that.

Comment (ryanurban) [reply wp-content/themes/core/assets/js/src/utils/dom/smooth-anchors.js:28]: Scope this: https://github.com/moderntribe/square-one/pull/677/files#diff-3322c350466fd62a27abec4bb2f60e0638b414251d64b083108843ccf413c699R37

We only apply this overall utility to anchors that have an anchor hash href value : )

Comment (ryanurban) [reply wp-content/themes/core/assets/js/src/utils/dom/smooth-anchors.js:28]: Oh derp, ignore me, right, the thing it might reference may not be focusable. Haha, what I'm doing up for a11y will also address that too : )

Comment (tjdunklee) [reply wp-content/themes/core/assets/js/src/utils/dom/smooth-anchors.js:28]: No problem at all! Great! Looking forward to seeing what you come up with.

**Comment by ryanurban** [comment]
@tjdunklee scope this and let me know what you think and if it makes sense. I do wonder if we should keep/have the auto focus option for the smooth scroll functionality? Would there ever be a case where we wouldn't want to add focus to the targeted element? My gut, for a11y sake, says no. What do you think?

**Comment by ryanurban** [comment]
My additional fixes the browser issue where skip links don't work correctly. As in you click on the skip link (or any anchor hash for that matter), it positions you to the correct placement on the page, but doesn't shift focus as it should to the element that you've skipped to and you wind up very annoyingly tabbing back up at the top of the site.

This fixes it across the board and handles adding/removing of tabindex for the targeted element based on whether or not it's a focusable item. I also added a custom event to ensure this works when smooth scroll is active on a project as due to how we handle that (preventing the default behavior and then adding the hash with pushstate), the `hashchange` event doesn't get triggered correctly otherwise. I figured it'd be nice to handle the focus management in one place.

**Comment by tjdunklee** [comment]
Everything is looking good to me @ryanurban! I gave it a quick test locally visually and also with VoiceOver. I like it. Do you want to update the changelog before we merge in?

**Pull request closed without merging**
